grabbit/src/main/groovy/com/twcable/grabbit/jcr/ProtoNodeDecorator.groovy

```diff
@@ -0,0 +1,90 @@
+package com.twcable.grabbit.jcr
+
+import com.twcable.grabbit.proto.NodeProtos.Node as ProtoNode
+import com.twcable.grabbit.proto.NodeProtos.Value as ProtoValue
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.apache.jackrabbit.commons.JcrUtils
+
+import javax.annotation.Nonnull
+import javax.jcr.Node as JCRNode
+import javax.jcr.Session
+
+import static org.apache.jackrabbit.JcrConstants.JCR_MIXINTYPES
+import static org.apache.jackrabbit.JcrConstants.JCR_PRIMARYTYPE
+
+@CompileStatic
+@Slf4j
+class ProtoNodeDecorator {
+
+    @Delegate
+    ProtoNode innerProtoNode
+
+    Collection<ProtoPropertyDecorator> protoProperties
+
+
+    ProtoNodeDecorator(@Nonnull ProtoNode node) {
+        if(!node) throw new IllegalArgumentException("node must not be null!")
+        this.innerProtoNode = node
+        this.protoProperties = node.properties.propertyList.collect { new ProtoPropertyDecorator(it) }
```

Comment:
All nodes have at least one property (jcr:primaryType), and per Jackrabbit, likely many more